docs(react-grid): add reference for createRowCache function (#2069)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/reference/create-row-cache.md

# createRowCache

A function that instantiates a default implementation of row cache. The cache is advised to be used in [lazy load mode](../guides/lazy-loading.md) to reduce the number of network requests.

## User reference

### Arguments

Name | Type | Default | Description
-----|------|---------|------------
pageSize | number | 100 | The number of rows on virtual page. It must be equal to the value of pageSize property of the [VirtualTableState](virtual-table-state.md#properties) plugin.
capacity | number | Infinity | The maximum number of pages that cache should store. When the reaches its capacity, it discards the least recently used virtual page.

### Return value

Type | Description
-----|------------

> RowCache | A row cache.

Reviewer: Can we add a reference to RowCache interface?
Can we add a section `How to import`?

Author: It's a good idea to add a usage example.
The RowCache interface described below, I think it's better to keep them in a single document

Reviewer: No, I mean other points.

- Add [RowCache](`#RowCache`) link.
- Add import section as we did here
![image](https://user-images.githubusercontent.com/15093892/59030145-d13e2d00-8868-11e9-869b-f07ef0d3a12f.png)

# createRowCache

> A function that instantiates a default implementation of row cache. The cache is advised to be used in [lazy load mode](../guides/lazy-loading.md) to reduce the number of network requests.

Reviewer 2: ```suggestion
A function that instantiates the default implementation of row caching. We recommend that you use row caching in [lazy loading mode](../guides/lazy-loading.md) to reduce the number of network requests.
```

## Import

Use the following statement to import the function:

```js
import { createRowCache } from '@devexpress/dx-react-grid';
```

## User reference

### Arguments

Name | Type | Default | Description
-----|------|---------|------------

> pageSize | number | 100 | The number of rows on virtual page. It must be equal to the value of pageSize property of the [VirtualTableState](virtual-table-state.md#properties) plugin.

Reviewer 2: ```suggestion
pageSize | number | 100 | The number of rows on a virtual page. It should be equal to the `pageSize` property value specified in the [VirtualTableState](virtual-table-state.md#properties) plugin.
```

> capacity | number | Infinity | The maximum number of pages that cache should store. When the cache reaches its capacity, it discards the least recently used virtual page.

Reviewer 2: ```suggestion
capacity | number | Infinity | The maximum number of virtual pages that the cache should store. When the capacity is reached, the least recently used page is removed from the cache to make room for new pages.
```

> ### Return value

Reviewer 2: ```suggestion
### Return Value
```

> capacity | number | Infinity | The maximum number of virtual pages that the cache should store. When the capacity is reached, the least recently used page is removed from the cache to make room for new pages.

Reviewer 3: ```suggestion
capacity | number | Infinity | The maximum number of virtual pages that the cache should store. When capacity is reached, the oldest used page is removed from the cache to make room for new pages.
```

Type | Description
-----|------------
[RowCache](#rowcache) | A row cache.

## Interfaces

### RowCache

Describes a row cache.

Field | Type | Description
------|------|------------

> getRows | (skip: number, take: number) => any[] | Returns `take` number of rows starting with the `skip` row from the cache.

Reviewer 2: ```suggestion
getRows | (skip: number, take: number) => any[] | Returns from the cache the `take` number of rows starting with the `skip` row.
```

setRows | (skip: number, rows: ReadonlyArray&lt;any&gt;) => void | Adds `rows` to the cache.

> invalidate | () => void | Clears the cache

Reviewer 2: ```suggestion
invalidate | () => void | Clears the cache.
```
